src/libs/Network.js

```diff
@@ -160,6 +171,11 @@ function processNetworkRequestQueue() {
             .catch(error => onError(queuedRequest, error));
     });
 
+    // User came back online and the request queue will resume, thus we should clear the NETWORK_REQUEST_QUEUE
+    if (ActiveClientManager.isClientTheLeader() && retryRequestsAdded) {
```

Comment:
I'm still a little lost on why we only want to clear the queue when the client is the leader and there is also no explanation about the significance of `retryRequestsAdded`. Can you try explaining this better in the comments?

>// User came back online

How exactly does this tell us the user came back online?

Reply:
So When a user goes offline we filter the network request queue and put the retrievable requests on persisted Queue. But it is possible that a user can come back online without refreshing the page. Now those retriable requests will process as a part of `networkRequestQueue`. Thus we don't want to process the persisted queue `ONYXKEYS.NETWORK_REQUEST_QUEUE`. It should only run when the user refresh the page.

Reply:
We only allow the leader to handle the `ONYXKEYS.NETWORK_REQUEST_QUEUE` and make sure that we are only clear the queue when are processing those requests via normal `networkRequestQueue`. As we processed the normal queue, we will not need them to be persisted.

Reply:
Ok, it's still a little unclear to me, but I might have to just look harder at this and try to offer some better suggestions.